services/motion/builtin/replanner.go

```diff
@@ -0,0 +1,39 @@
+package builtin
+
+import (
+	"context"
+	"time"
+)
+
+type replanResponse struct {
+	err    error
+	replan bool
+}
+
+type replanner struct {
+	period       time.Duration
+	fn           func(ctx context.Context) replanResponse
```

Comment:
[q] in my understanding `replanner.fn()` is what will execute position polling, and obstacle detection.
I believe that to accomplish position polling we need the movementsensor.
With the way this is setup, how will we have access to that?

A similar question exists for how we will get the needed information to execute obstacle detection

Reply:
spoke offline - answer is we can define the variables above and use them